packages/blocks/src/api/registration.js

```diff
@@ -154,7 +162,7 @@ export function unstable__bootstrapServerSideBlockDefinitions( definitions ) {
 export function registerBlockType( name, settings ) {
 	settings = {
 		name,
-		...DEFAULT_BLOCK_TYPE_SETTINGS,
+		...cloneDeep( DEFAULT_BLOCK_TYPE_SETTINGS ),
```

Comment:
I'm not sure clone deep is the best solution here, especially for the icon. Could shallow cloning the settings and attributes work? I'm also not against moving it to the function.

Reply:
I've tried moving it directly to the function. Otherwise, we would need to clone deeply some keys of the defaults object and I think that might get unnecessary complex for relatively simple operation.